### PR TITLE
Implement VMI Replicaset for Applications

### PR DESCRIPTION
**Pull request by naiming-zededa**
- to allow VMI migration during node isolation or down
 - code to handle extra suffix for vmi name

 - Set nodename to lowercase when querying for pods Also check for hasprefix for the contVMIname also

### CODE REVIEW

File: pkg/pillar/hypervisor/kubevirt.go [hunk 1]
```diff
@@ -486,46 +514,48 @@ func (ctx kubevirtContext) Start(domainName string) error {
...
 		logrus.Errorf("couldn't get the kubernetes client API config: %v", err)
 		return err
 	}
 
-	// Create the VM
+	// Create the VMI ReplicaSet
 	i := 5
 	for {
-		_, err = virtClient.VirtualMachineInstance(kubeapi.EVEKubeNameSpace).Create(context.Background(), vmi)
+		_, err = virtClient.ReplicaSet(kubeapi.EVEKubeNameSpace).Create(repvmi)
```
Comment (andrewd-zededa) [inline pkg/pillar/hypervisor/kubevirt.go:537]: @naiming-zededa does this move to some default context?

**Review by andrewd-zededa — COMMENTED** [review]
mostly lgtm, just a question about context

File: pkg/pillar/hypervisor/kubevirt.go (hunk 1 above)
Comment (naiming-zededa) [reply pkg/pillar/hypervisor/kubevirt.go:537]: @andrewd-zededa, No, it's still in the 'eve-kube-app', is this you meant by context?

Comment (andrewd-zededa) [reply pkg/pillar/hypervisor/kubevirt.go:537]: no not namespace, the new Create() call removes context.Background() from the first arg

Comment (andrewd-zededa) [reply pkg/pillar/hypervisor/kubevirt.go:537]: I see on that version there is a default context.Background().
https://github.com/kubevirt/client-go/blob/09d24236db4ccb7a1872b8611d4baddd26a04ab3/kubecli/replicaset.go#L103

Comment (andrewd-zededa) [reply pkg/pillar/hypervisor/kubevirt.go:537]: the version we're on currently 1.2.0 https://github.com/naiming-zededa/eve/blob/01b9114736ccebbc607c221447b3f62e2d22b442/pkg/pillar/go.mod#L65

**Review by andrewd-zededa — APPROVED** [review]
LGTM

File: pkg/pillar/hypervisor/kubevirt.go (hunk 1 above)
Comment (naiming-zededa) [reply pkg/pillar/hypervisor/kubevirt.go:537]: well, they don't provide a way to pass in, so just default behavior of the context

**Pull request merged**